### PR TITLE
Fix Sessions Test App with Fake Subscriber

### PR DESCRIPTION
**Pull request by samedson**
Adds a fake subscriber SDKs to our test app with data collection enabled

### CODE REVIEW

**Review by visumickey — APPROVED** [review]
Nice. Let's make sure the hack does not stay for long! But a good solution.

File: FirebaseSessions/Tests/TestApp/Shared/MockSubscriberSDK.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+@objc(FIRMockSubscriberSDK) final class MockSubscriberSDK: NSObject, Library, SessionsSubscriber,
+  MockSubscriberSDKProtocol {
+  static func addDependency() {
+    FirebaseApp.registerInternalLibrary(
+      MockSubscriberSDK.self,
+      withName: "mock-firebase-sessions-subscriber-sdk"
+    )
+    SessionsDependencies.addDependency(name: SessionsSubscriberName.Performance)
```
Comment (leotianlizhan) [inline FirebaseSessions/Tests/TestApp/Shared/MockSubscriberSDK.swift:34]: So this mock sdk is pretending to be Fireperf right?

File: FirebaseSessions/Tests/TestApp/Shared/AppQualityDevAppApp.swift [hunk 2]
```diff
@@ -15,10 +15,13 @@
 
 import SwiftUI
 import FirebaseCore
+import FirebaseSessions
 
 @main
-struct AppQualityDevAppApp: App {
-  init() {
+class AppQualityDevAppApp: App {
+  required init() {
```
Comment (leotianlizhan) [inline FirebaseSessions/Tests/TestApp/Shared/AppQualityDevAppApp.swift:22]: I remember when talking about how session dependency works we said the dependencies would be added when loaded to memory, like `+ load()`, so is `required init()` equivalent to `+ load()`?

**Review by leotianlizhan — APPROVED** [review]
LGTM

File: FirebaseSessions/Tests/TestApp/Shared/MockSubscriberSDK.swift (hunk 1 above)
Comment (samedson) [reply FirebaseSessions/Tests/TestApp/Shared/MockSubscriberSDK.swift:34]: That's right

File: FirebaseSessions/Tests/TestApp/Shared/AppQualityDevAppApp.swift (hunk 2 above)
Comment (samedson) [reply FirebaseSessions/Tests/TestApp/Shared/AppQualityDevAppApp.swift:22]: So I'm cheating a little bit. Swift doesn't have the same `load` function (at least I couldn't figure it out in 5 mins of Googling), so I just call this function at the start up the app, but before the Firebase SDKs start up.

So technically this solution is called later than `load`, but it's earlier than Firebase Initialization `FirebaseApp.configure()`, so it's ok.

Comment (leotianlizhan) [reply FirebaseSessions/Tests/TestApp/Shared/AppQualityDevAppApp.swift:22]: Got it, makes sense!

**Pull request merged**